inc/toolbox.class.php

```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __("Fileinfo extension of your parser PHP is not installed"),
+            'class'     => 'finfo'
+         ],
+         'json'     => [
+            'title'     => __('Test json functions'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __("GLPI can't work correctly without the json_encode and json_decode functions"),
+            'functions' => [
+               'json_encode',
+               'json_decode'
+            ]
+         ],
+         'mbstring' => [
+            'title'     => __('Mbstring extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Mbstring extension of your parser PHP is not installed')
+         ],
+         'gd'       => [
+            'title'     => __('GD extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('GD extension of your parser PHP is not installed')
+         ],
+         'zlib'     => [
+            'title'     => __('Zlib extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Zlib extension of your parser PHP is not installed')
+         ],
+         'curl'      => [
+            'title'     => __('Curl extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Curl extension of your parser PHP is not installed')
+         ]
+      ];
+
+      $loaded_extensions = [];
+      //check for PHP extensions
+      foreach ($needed_extentions as $ext => $params) {
+         $success = true;
+
+         if (isset($params['functions'])) {
+            foreach ($params['functions'] as $function) {
+               if (!function_exists($func)) {
+                  $success = false;
+               }
+            }
+         } elseif (isset($param['class'])) {
+            if (!class_exists($params['class'])) {
+               $success = false;
+            }
+         } else {
+            if (extension_loaded($ext)) {
+               $success = false;
+            }
+         }
 
-      } else {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ok_min.png' alt=\"".
-                     __s('The functionality is found - Perfect!'). "\" title=\"".
-                     __s('The functionality is found - Perfect!')."\"></td>";
+         echo "<tr class=\"tab_bg_1\"><td class=\"left b\">{$params['title']}</td>";
+         if ($success === true) {
+            echo "<td><img src=\"{$CFG_GLPI['root_doc']}/pics/ok_min.png\"
+                    alt=\"{$params['success']}\"
+                    title=\"{$params['success']}\"></td>";
+         } else {
+            echo "<td class=\"red\"><img src=\"{$CFG_GLPI['root_doc']}/pics/ko_min.png\">{$params['error']}</td>";
+            $error = 2;
```

Comment:
See previous.

if ($error=1) ....'you should install '
if ($error=2) ....'you MUST install '